avm/ptn/data/private-analytical-workspace/tests/e2e/uc02-priv/main.test.bicep

```diff
@@ -0,0 +1,114 @@
+targetScope = 'subscription'
+
+metadata name = 'Use Case 2 - fully private'
+metadata description = 'Deployment in an Existing, Enterprise-Specific Virtual Network - fully private.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-data-privateanalyticalworkspace-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'dpawuc02priv'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+var subnetName01 = 'private-link-subnet'
+var subnetName02 = 'dbw-frontend-subnet'
+var subnetName03 = 'dbw-backend-subnet'
```

Comment:
if you want to align this to other tests, I'd recommend to hardcode these values in the subnets of the dependencies.bicep file directly as they cannot conflict with other resources. This possiblity of a conflict is actualy the main reason why we ask contributors to generate all top-level resource names in this file and pass it into the dependencies.bicep file. However, for anything that is then internal to those resources we don't require it anymore and it frankly makes the file easier if minimize the amount of parmaeters and keep the sub-resource names close to where they are used 😉 
No requirement, but a recommendation.

Reply:
Yes, make sense. Refactored dependencies.bicep files.

Reply:
Hey @jbinko,
thanks for that. Unfortunately we went a 'bit' too far in the opposite direction now 😄 Let me elaborate on my initial comment a bit more:
What we usually suggest is to have the names of all top-level resources (vnet, key vault, law, you name it) in the main.test.bicep, where the name is created and the values are passed into the `dependencies.bicep`. For example like here: 
https://github.com/Azure/bicep-registry-modules/blob/b673ec8840fa9eee5e71fa76bf991b0899d5714d/avm/res/key-vault/vault/tests/e2e/max/main.test.bicep#L34-L42

In the above example, you can also see the main reason we do that: We inject the `namePrefix` & `serviceShort` in each resource name + we add a `dep-` prefix to all resources. 
So you may wonder why:
- The `namePrefix` & `serviceShort` are here because their combination is always different from fork to fork and test case to test case. So there won't be any conflicts with other contributors, no you without yourself when running multiple pipelines
- Also, the `dep-` prefix ensures that our WAF-aligned tests (via PSRule) ignore these resources when testing. Reason being: We want PSRule to test you module, not your module's temporary test dependencies.
- (Bonus): Because the names of the top-level resources are generated in the `main.test.bicep` file it's a 'bit' easier for troubleshooting as you can immediately see where a name an error message may contain comes from. Not super important, but a nice side-effect.

Now the one thing we don't have here are the nested resources. Like the subnet names I called out, or let's say the names of secrets in a key vault. As those are 'inner' resources to a 'main' resource, they usually never conflict with anything. These are hence the values we expect contributors to hardcode in their `dependencies.bicep` files als none of the previously mentioned benefits apply for them.

All in all, it keeps both your `main.test.bicep` & `depepdencies.bicep` file(s) quite distinct in scope and easy to understand (or that's at least the idea 😄). I hope the above made sense.